File: agent/commands-generator-api/src/main/java/org/matilda/commands/PluginDependencies.java
package org.matilda.commands;

import javax.inject.Inject;

public class PluginDependencies {
    public final CommandRegistryManager commandRegistryManager;

    public final CommandRunner commandRunner;

    @Inject
    public PluginDependencies(CommandRegistryManager commandRegistryManager, CommandRunner commandRunner) {
        this.commandRegistryManager = commandRegistryManager;
        this.commandRunner = commandRunner;
    }
}

File: agent/commands-generator-api/src/main/java/org/matilda/commands/PluginDependencies.kt
package org.matilda.commands

import java.util.*
import javax.inject.Inject

class PluginDependencies @Inject constructor() {
    @Inject
    lateinit var commandRegistryManager: CommandRegistryManager

    @Inject
    lateinit var commandRunner: CommandRunner

    @Inject
    lateinit var random: Random
}
